### PR TITLE
add some internal sql apis

### PR DESCRIPTION
**Pull request by mairaw**
This is needed for APIscan.

A lot of these existed prior in MSDN (e.g. https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/bb882618%28v%3dvs.100%29)

Made some changes according to our API style guide and also made some fixes (for example, there's no get_Stream method - that's just the getter for the Stream property).

@douglaslMS put you as the author since this is in the SQL arena.

### CODE REVIEW

File: docs/framework/additional-apis/toc.md [hunk 1]
```diff
@@ -1,4 +1,22 @@
 # [Additional class libraries and APIs](index.md)
+## Microsoft.SqlServer.Server
+### SmiOrderProperty Class
```
Comment (rpetrusha) [inline docs/framework/additional-apis/toc.md]: Should there be separate pages for each type (SmiOrderPrpoerty, SqlStreamChars), and SqlChars link to the managed reference type?

Comment (rpetrusha) [inline docs/framework/additional-apis/toc.md]: Should there be separate pages for each type (SmiOrderPrpoerty, SqlStreamChars), and SqlChars link to the managed reference type?

**Review by rpetrusha — APPROVED** [review]
I've left a number of suggestions and one comment, @mairaw. Once you review them, this is ready to merge.

File: docs/framework/additional-apis/toc.md (hunk 1 above)
Comment (mairaw) [reply docs/framework/additional-apis/toc.md]: I used to do that for the namespace and class and I thought it was a weird behavior to go to a completely different place.

Comment (mairaw) [reply docs/framework/additional-apis/toc.md]: Not sure. I'm adding the ones that were failing apiscan.

**Comment by mairaw** [comment]
Thanks @rpetrusha for your always thorough review. I've addressed most of your comments except for the toc ones where I've left some explanation.

**Pull request merged**